ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/HardFork/Combinator/InjectTxs.hs

```diff
@@ -295,4 +221,4 @@ matchValidatedTxsNS ::
   -> NS f xs
   -> [NS WrapValidatedGenTx xs]
   -> ([Mismatch WrapValidatedGenTx f xs], NS (Product f ([] :.: WrapValidatedGenTx)) xs)
-matchValidatedTxsNS = matchPolyTxsNS
+matchValidatedTxsNS ips ns txs = bimap (map snd) Telescope.tip $ matchPolyTxsTele ips (Telescope.fromTip ns) txs
```

Comment:
Why are we using a telescope here if we're just discarding it again?

Reply:
Because the underlying `matchPolyTxsTele` uses a telescope.

It has to do so because need a telescope in the call site of that one alone (see Ouroboros\Consensus\HardFork\Combinator\Mempool).

We need to `hcizipWith` through a telescope.